monkey/common/event_queue/i_event_queue.py

```diff
@@ -0,0 +1,85 @@
+from abc import ABC, abstractstaticmethod
+from typing import Any, Callable, Sequence
+
+from common.events import AbstractEvent
+
+
+class IEventQueue(ABC):
+    """
+    Manages subscription and publishing of events
+    """
+
+    @abstractstaticmethod
+    def subscribe_all(subscriber: Callable[..., Any]):
+        """
+        Subscribes a subscriber to all events
+
+        :param subscriber: Callable that should subscribe to events
+        """
+
+        pass
+
+    @abstractstaticmethod
+    def subscribe_types(types: Sequence[AbstractEvent], subscriber: Callable[..., Any]):
```

Comment:
I think the most common case will be that a subscriber will subscribe to a single type. All events of the same type will have the same structure. This means that a subscriber could change because the type changes. That's fine. If a subscriber subscribes to multiple types, then it may change because any of the types change. In other words, a subscriber that subscribes to multiple types would likely be violating the Single Responsibility Principle.